_posts/2018-10-09-enabling-ecc-on-pc-engines-apu2.md

```diff
@@ -0,0 +1,354 @@
+2018-09-25 14:01:33 - [CONFIG] ECC injection: enabled
+2018-09-25 14:01:34 - [CONFIG] Auto mode: enabled
+2018-09-25 14:01:34 - [CONFIG] Skip splash screen: true
+2018-09-25 14:01:34 - [CONFIG] Console mode: 0
+2018-09-25 14:01:34 - Applying configurations
+2018-09-25 14:01:34 - [CONFIG] This platform has 1 logical processors of which 1 are enabled.
+2018-09-25 14:01:34 - Console size = 80 x 25
+2018-09-25 14:01:34 - Screen resolution is too low (800 x 600). Setting screen size to the minimum 1024 x 768
+2018-09-25 14:01:35 - Unable to set screen size
+2018-09-25 14:01:35 - Screen size = 800 x 600
+2018-09-25 14:01:35 - Char width=8 height=19
+2018-09-25 14:01:35 - Loading images
+2018-09-25 14:01:35 - AMD 10h and greater chipset init
+2018-09-25 14:01:35 - DRAM config low=83090000
+2018-09-25 14:01:35 - MCA NB config=C8700044
+2018-09-25 14:01:35 - MC4_CTL=0000000000000000
+2018-09-25 14:01:35 - MC4_CTL_MASK=0000000004000000
+2018-09-25 14:01:35 - MCG_CTL low=00000010
+2018-09-25 14:01:35 - *** TEST SESSION - 2018-09-25 14:01:35 ***
+2018-09-25 14:01:35 - CPU selection mode = 0
+2018-09-25 14:01:35 - Starting pass #1 (of 2)
+2018-09-25 14:01:35 - Running test #3 (Test 3 [Moving inversions, ones & zeroes])
+2018-09-25 14:01:35 - MtSupportRunAllTests - Injecting ECC error
+2018-09-25 14:01:35 - inject_amd64 - new nb_arr_add = 80000000
+2018-09-25 14:01:35 - inject_amd64 - new dram_ecc = 0012000F
+2018-09-25 14:01:35 - MCA NB Status High=00000000
+2018-09-25 14:01:35 - inject_amd64 - new nb_arr_add = 80000002
+2018-09-25 14:01:35 - inject_amd64 - new dram_ecc = 0012000F
+2018-09-25 14:01:35 - MCA NB Status High=00000000
+2018-09-25 14:01:35 - inject_amd64 - new nb_arr_add = 80000004
+2018-09-25 14:01:36 - inject_amd64 - new dram_ecc = 0012000F
+2018-09-25 14:01:36 - MCA NB Status High=00000000
+2018-09-25 14:01:36 - MtSupportRunAllTests - Setting random seed to 0x50415353
+2018-09-25 14:01:36 - MtSupportRunAllTests - Start time: 422 ms
+2018-09-25 14:01:36 - ReadMemoryRanges - Available Pages = 1035072
+2018-09-25 14:01:36 - MtSupportRunAllTests - Enabling memory cache for test
+2018-09-25 14:01:36 - MtSupportRunAllTests - Enabling memory cache complete
+2018-09-25 14:01:36 - Start memory range test (0x0 - 0x12F000000)
+2018-09-25 14:01:36 - Pre-allocating memory ranges >=16MB first...
+2018-09-25 14:01:36 - All memory ranges successfully locked
+2018-09-25 14:02:54 - Cleanup - Releasing all memory ranges...
+2018-09-25 14:02:54 - MtSupportRunAllTests - Test execution time: 78.372 (Test 3 cumulative error count: 0)
+(...)
+```
+
+As you can see, MemTest86 injects ECC errors (or at least tries to), but these
+errors are not reported, which means that something is wrong. There is a lot of
+relevant debug information included, but names differ slightly between log and
+register names in BKDG. More information about reported values and their impact
+on the issue will be revealed in the next sections.
```

Comment:
@krystian-hebel I think more robust description of what we seen in log is needed. Where we see injection, whyc we claim errors are not reported? Why we think this is wrong? At least briefly to not completly lost our reader.